File: qlack-be-rules/qlack-be-rules-ui/angular/src/app/dto/rule-dto.ts
import {BaseDto} from './base-dto';
import {CategoryDto} from './category-dto';
import {RuleVersionDto} from './rule-version-dto';

export interface RuleDto extends BaseDto {
  name?: string;
  description?: string;
  status?: boolean;
  ruleVersions?: RuleVersionDto[];
  categories?: CategoryDto[];
}

File: qlack-be-rules/qlack-be-rules-server/src/main/java/com/eurodyn/qlack/rulesdemo/dto/RuleDTO.java
package com.eurodyn.qlack.rulesdemo.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
public class RuleDTO extends BaseDTO {

    private String name;

    private String description;

    private boolean status;

    private List<RuleVersionDTO> ruleVersions;

    private List<CategoryDTO> categories;
}
